SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,32 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // If this node only has one non-dummy child node, that node is returned by this function; otherwise the function returns the node itself
+    func mergeNode() -> Node {
+        var nonDummyChildCount = 0
+        var nonDummyChildren : Node? = nil
+        for childNode in self.childNodes {
+            if childNode.dummy == false {
+                nonDummyChildCount = nonDummyChildCount + 1
+                nonDummyChildren = childNode
+            }
+        }
+        if nonDummyChildCount == 1 {
+            return nonDummyChildren!
+        }
+        else{
+            return self
+        }
+    }
+
+    // it will make every child node shrink if that child has only single nonDummy child node
+    func ShrinkChildren() {
```

Comment:
Please change this to camelCase: `shrinkChildren`. Only classes, structs, and enums should be in TitleCase: `Node`, `TouchGrid`, etc.